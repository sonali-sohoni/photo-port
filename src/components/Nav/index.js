import React from "react";
import { useState, useEffect } from "react";
import { capitalizeFirstLetter } from "../../utils/helpers";

const Nav = ({
	currentCategory,
	setCurrentCategory,
	categories,
	contactSelected,
	setContactSelected,
}) => {
	function categorySelected(cat) {
		console.log(`hello ${cat.name}`);
		setCurrentCategory(cat);
	}

	useEffect(() => {
		document.title = capitalizeFirstLetter(currentCategory.name);
	}, [currentCategory]); //causes rendering of the component

	return (
		<header>
			<a data-testid="link" href="/">
				<span role="img" aria-label="camera">
					ðŸ“¸
				</span>
				Oh Snap!
			</a>
			<nav>
				<ul className="flex-row">
					<li className="mx-2">
						<a
							data-testid="about"
							href="#about"
							onClick={() => setContactSelected(false)}
						>
							About Me
						</a>
					</li>
					<li className={`mx-2 ${contactSelected && "navActive"}`}>
						<span onClick={() => setContactSelected(true)}>Contact</span>
					</li>
					{categories.map((category) => (
						<li
							className={`mx-1
            ${
							category.name === currentCategory.name &&
							!contactSelected &&
							"navActive"
						}`}
							key={category.name}
						>
							<span
								onClick={() => {
									categorySelected(category);
									contactSelected(false);
								}}
							>
								{capitalizeFirstLetter(category.name)}
							</span>
						</li>
					))}
				</ul>
			</nav>
		</header>
	);
};
export default Nav;
