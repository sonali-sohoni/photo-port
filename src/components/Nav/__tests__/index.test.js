import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

afterEach(cleanup);

describe("Nav Component", () => {
	it("renders", () => {
		render(<Nav />);
	});

	it("Snapshot test for Nav component", () => {
		const { asFragment } = render(<Nav />);
		expect(asFragment()).toMatchSnapshot();
	});
});

describe("camera emoji is visible", () => {
	it("inserts emoji in h2", () => {
		const { getByLabelText } = render(<Nav />);

		expect(getByLabelText("camera")).toHaveTextContent("ðŸ“¸");
	});
});

describe("Links are visible", () => {
	it("insert text into links", () => {
		//get the function using render(<component/>)
		const { getByTestId } = render(<Nav />);
		expect(getByTestId("link")).toHaveTextContent("Oh Snap!");
		expect(getByTestId("about")).toHaveTextContent("About Me");
	});
});
