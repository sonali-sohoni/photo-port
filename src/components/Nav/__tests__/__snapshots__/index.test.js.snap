// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component Snapshot test for Nav component 1`] = `
<DocumentFragment>
  <header>
    <a
      data-testid="link"
      href="/"
    >
      <span
        aria-label="camera"
        role="img"
      >
        ðŸ“¸
      </span>
      Oh Snap!
    </a>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About Me
          </a>
        </li>
        <li
          class="mx-2"
        >
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            commercial
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            portraits
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            food
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            landscape
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
